src/Components/VimTree/Component_VimTree.re

```diff
@@ -271,9 +272,28 @@ let update = (msg, model) => {
 
     switch (outmsg) {
     | Component_VimList.Nothing => (model, Nothing)
+    | Component_VimList.Touched({index}) =>
+      switch (Component_VimList.get(index, treeAsList)) {
+      | Some(ViewLeaf({data, _})) => (model, Touched(data))
+      // TODO: Expand / collapse
+      | Some(ViewNode({data, expanded, _})) =>
+        let expansionContext =
+          expanded
+            ? model.expansionContext
+              |> ExpansionContext.collapse(data.uniqueId)
+            : model.expansionContext |> ExpansionContext.expand(data.uniqueId);
+
+        (
+          updateTreeList(model.trees, expansionContext, model),
+          expanded ? Collapsed(data.inner) : Expanded(data.inner),
+        );
+
```

Comment:
Is there a reason we can't use the `toggleExpanded` function here? Seems like it would remove some duplication

Reply:
We could if it fit here. I missed that we already have this function.